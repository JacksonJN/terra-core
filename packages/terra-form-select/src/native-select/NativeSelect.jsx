import React, {
  useState,
  useRef,
} from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import classNamesBind from 'classnames/bind';
import { injectIntl, intlShape } from 'react-intl';
import ThemeContext from 'terra-theme-context';
import {
  defaultPlaceholderValue,
  isCurrentPlaceholder,
  getDisplay,
  getFirstValue,
} from './NativeUtils';
import {
  optionPropType,
  optGroupPropType,
} from './NativePropTypes';

import styles from './NativeSelect.module.scss';

const cx = classNamesBind.bind(styles);

const propTypes = {
  /**
   * The string containing ids for elements to describe the select.
   */
  ariaDescribedBy: PropTypes.string,
  /**
   * The aria label string value for the select.
   */
  ariaLabel: PropTypes.string.isRequired,
  /**
   * Additional attributes to spread onto the select.
   */
  // eslint-disable-next-line react/forbid-prop-types
  attrs: PropTypes.object,
  /**
   * The default value of the select. Can be a string, or number.
   */
  defaultValue: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  /**
   * Whether the input is disabled.
   */
  disabled: PropTypes.bool,
  /**
   * The Select identifier. Links the htmlFor of the field to the select identifier.
   */
  id: PropTypes.string.isRequired,
  /**
   * @private
   * The intl object to be injected for translations.
   */
  intl: intlShape.isRequired,
  /**
   * Whether the input is invalid.
   */
  isInvalid: PropTypes.bool,
  /**
   * Whether the field displays as Incomplete. Use when no value has been provided. _(usage note: `required` must also be set)_.
   */
  isIncomplete: PropTypes.bool,
  /**
   * Callback function triggered when the select value changes. function(value)
   */
  onChange: PropTypes.func,
  /**
   * The array of select options and opt groups.
   */
  options: PropTypes.arrayOf(PropTypes.oneOfType([optionPropType, optGroupPropType])),
  /**
   * Placeholder data. The presence of this data object is used as the indicator for whether or not a placeholder is used.
   */
  placeholder: PropTypes.shape({
    /**
     * Whether the placeholder is included as an option to clear selection.
     */
    allowClear: PropTypes.bool,
    /**
     * Optional override of the default display.
     */
    display: PropTypes.string,
    /**
     * Optional override of the default value.
     */
    value: PropTypes.string,
  }),
  /**
   * Whether the field is required.
   */
  required: PropTypes.bool,
  /**
   * The value of the select. Can be a string or number.
   */
  value: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
};

const defaultProps = {
  disabled: false,
  isIncomplete: false,
  isInvalid: false,
  options: [],
  required: false,
};

const createPlaceholder = (placeholder, intl) => {
  if (!placeholder) {
    return undefined;
  }
  const display = placeholder.display ? placeholder.display : intl.formatMessage({ id: 'Terra.form.select.defaultDisplay' });
  const value = placeholder.value ? placeholder.value : defaultPlaceholderValue;
  const attrs = placeholder.allowClear ? {} : { disabled: true, hidden: true };
  return <option value={value} {...attrs}>{display}</option>;
};

const createOptions = options => options.map(current => {
  if (current.childOptions) {
    return (
      <optgroup key={`${current.display} ${current.childOptions.length}`} label={current.display}>
        {createOptions(current.childOptions)}
      </optgroup>
    );
  }
  return <option key={`${current.value}`} value={current.value}>{current.display}</option>;
});

const NativeSelect = ({
  ariaDescribedBy,
  ariaLabel,
  attrs,
  disabled,
  defaultValue,
  id,
  intl,
  isInvalid,
  isIncomplete,
  onChange,
  options,
  placeholder,
  required,
  value,
  ...customProps
}) => {
  const [currentValue, setCurrentValue] = useState(defaultValue || getFirstValue(options, placeholder));
  const refIsControlled = useRef(!defaultValue && value);
  const refSelect = useRef();
  const theme = React.useContext(ThemeContext);

  const handleOnMouseDown = () => {
    refSelect.current.setAttribute('data-focus-interaction', 'mouse');
  };
  const handleOnBlur = () => {
    refSelect.current.setAttribute('data-focus-interaction', 'none');
  };
  const handleOnFocus = () => {
    if (refSelect.current.getAttribute('data-focus-interaction') !== 'mouse') {
      refSelect.current.setAttribute('data-focus-interaction', 'keyboard');
    }
  };
  const handleOnChange = event => {
    if (onChange) {
      onChange(event);
    }
    if (!refIsControlled.current) {
      setCurrentValue(event.currentTarget.value);
    }
  };

  let controlledValue;
  if (refIsControlled.current) {
    controlledValue = value || getFirstValue(options, placeholder);
  }

  const selectAttrs = {
    'aria-describedby': ariaDescribedBy,
    'aria-label': ariaLabel,
    id,
    disabled,
    'aria-invalid': isInvalid || undefined,
    required,
    value: controlledValue || currentValue,
  };

  const nativeClassNames = classNames(
    cx(
      'native',
      theme.className,
      { disabled },
      { invalid: isInvalid },
      { incomplete: required && isIncomplete },
      { placeholder: isCurrentPlaceholder(selectAttrs.value, placeholder) },
    ),
    customProps.className,
  );

  return (
    <div
      {...customProps}
      className={nativeClassNames}
      ref={refSelect}
      data-focus-interaction="none"
    >
      <div aria-hidden className={cx('frame')}>
        <div
          className={cx('display')}
          aria-disabled={disabled || undefined}
        >
          {getDisplay(selectAttrs.value, options, placeholder, intl)}
        </div>
        <div className={cx('arrow')}>
          <div className={cx('arrow-icon')} />
        </div>
      </div>
      <select
        {...attrs}
        {...selectAttrs}
        className={cx('select')}
        onChange={handleOnChange}
        onMouseDown={handleOnMouseDown}
        onFocus={handleOnFocus}
        onBlur={handleOnBlur}
      >
        {createPlaceholder(placeholder, intl)}
        {createOptions(options)}
      </select>
    </div>
  );
};

NativeSelect.propTypes = propTypes;
NativeSelect.defaultProps = defaultProps;

export default injectIntl(NativeSelect);
