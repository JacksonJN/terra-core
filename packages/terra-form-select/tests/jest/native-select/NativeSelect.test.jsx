import React from 'react';
/* eslint-disable-next-line import/no-extraneous-dependencies */
import { mountWithIntl } from 'terra-enzyme-intl';
import NativeSelect from '../../../src/native-select/NativeSelect';

describe('NativeSelect', () => {
  it('should render a default NativeSelect', () => {
    const wrapper = mountWithIntl(
      <NativeSelect
        id="test-id"
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a NativeSelect with an aria label', () => {
    const wrapper = mountWithIntl(
      <NativeSelect
        id="test-id"
        ariaLabel="test-label"
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a NativeSelect aria described by', () => {
    const wrapper = mountWithIntl(
      <NativeSelect
        id="test-id"
        ariaDescribedBy="test-described-id"
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a NativeSelect with options', () => {
    const wrapper = mountWithIntl(
      <NativeSelect
        id="test-id"
        options={[
          { value: 'volvo', display: 'Volvo' },
          { value: 'saab', display: 'Saab' },
          { value: 'mercedes', display: 'Mercedes' },
          { value: 'audi', display: 'Audi' },
        ]}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a NativeSelect with options and defaultValue', () => {
    const wrapper = mountWithIntl(
      <NativeSelect
        id="test-id"
        defaultValue="saab"
        options={[
          { value: 'volvo', display: 'Volvo' },
          { value: 'saab', display: 'Saab' },
          { value: 'mercedes', display: 'Mercedes' },
          { value: 'audi', display: 'Audi' },
        ]}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a NativeSelect with options and value', () => {
    const wrapper = mountWithIntl(
      <NativeSelect
        id="test-id"
        value="audi"
        options={[
          { value: 'volvo', display: 'Volvo' },
          { value: 'saab', display: 'Saab' },
          { value: 'mercedes', display: 'Mercedes' },
          { value: 'audi', display: 'Audi' },
        ]}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a NativeSelect with onChange', () => {
    const wrapper = mountWithIntl(
      <NativeSelect
        id="test-id"
        onChange={jest.fn()}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a NativeSelect with a default placeholder', () => {
    const wrapper = mountWithIntl(
      <NativeSelect
        id="test-id"
        placeholder={{}}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a NativeSelect with a custom placeholder', () => {
    const wrapper = mountWithIntl(
      <NativeSelect
        id="test-id"
        placeholder={{ display: 'test-placeholder-display', value: 'test-placeholder-value' }}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a NativeSelect allowing clear', () => {
    const wrapper = mountWithIntl(
      <NativeSelect
        id="test-id"
        placeholder={{ allowClear: true }}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a NativeSelect as invalid', () => {
    const wrapper = mountWithIntl(
      <NativeSelect
        id="test-id"
        isInvalid
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a NativeSelect as disabled', () => {
    const wrapper = mountWithIntl(
      <NativeSelect
        id="test-id"
        disabled
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a NativeSelect with opt groups', () => {
    const wrapper = mountWithIntl(
      <NativeSelect
        id="test-id"
        options={[
          {
            display: 'Swedish Cars',
            childOptions: [
              { value: 'volvo', display: 'Volvo' },
              { value: 'saab', display: 'Saab' },
            ],
          },
          {
            display: 'German Cars',
            childOptions: [
              { value: 'mercedes', display: 'Mercedes' },
              { value: 'audi', display: 'Audi' },
            ],
          },
        ]}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a NativeSelect required', () => {
    const wrapper = mountWithIntl(
      <NativeSelect
        id="test-id"
        required
        isIncomplete
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a NativeSelect isIncomplete', () => {
    const wrapper = mountWithIntl(
      <NativeSelect
        id="test-id"
        required
        isIncomplete
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a NativeSelect required and isIncomplete', () => {
    const wrapper = mountWithIntl(
      <NativeSelect
        id="test-id"
        required
        isIncomplete
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
