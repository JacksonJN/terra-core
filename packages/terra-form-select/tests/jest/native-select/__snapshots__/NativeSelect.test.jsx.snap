// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NativeSelect should render a NativeSelect allowing clear 1`] = `
<NativeSelect
  disabled={false}
  id="test-id"
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
  invalid={false}
  isIncomplete={false}
  options={Array []}
  placeholder={Object {}}
  required={false}
/>
`;

exports[`NativeSelect should render a NativeSelect aria described by 1`] = `
<NativeSelect
  ariaDescribedBy="test-described-id"
  disabled={false}
  id="test-id"
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
  invalid={false}
  isIncomplete={false}
  options={Array []}
  required={false}
/>
`;

exports[`NativeSelect should render a NativeSelect as disabled 1`] = `
<NativeSelect
  disabled={true}
  id="test-id"
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
  invalid={false}
  isIncomplete={false}
  options={Array []}
  required={false}
/>
`;

exports[`NativeSelect should render a NativeSelect as invalid 1`] = `
<NativeSelect
  disabled={false}
  id="test-id"
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
  invalid={true}
  isIncomplete={false}
  options={Array []}
  required={false}
/>
`;

exports[`NativeSelect should render a NativeSelect isIncomplete 1`] = `
<NativeSelect
  disabled={false}
  id="test-id"
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
  invalid={false}
  isIncomplete={true}
  options={Array []}
  placeholder={Object {}}
  required={true}
/>
`;

exports[`NativeSelect should render a NativeSelect required 1`] = `
<NativeSelect
  disabled={false}
  id="test-id"
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
  invalid={false}
  isIncomplete={true}
  options={Array []}
  placeholder={Object {}}
  required={true}
/>
`;

exports[`NativeSelect should render a NativeSelect required and isIncomplete 1`] = `
<NativeSelect
  disabled={false}
  id="test-id"
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
  invalid={false}
  isIncomplete={true}
  options={Array []}
  placeholder={Object {}}
  required={true}
/>
`;

exports[`NativeSelect should render a NativeSelect with a placeholder 1`] = `
<NativeSelect
  disabled={false}
  id="test-id"
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
  invalid={false}
  isIncomplete={false}
  options={Array []}
  placeholder={Object {}}
  required={false}
/>
`;

exports[`NativeSelect should render a NativeSelect with an aria label 1`] = `
<NativeSelect
  ariaLabel="test-label"
  disabled={false}
  id="test-id"
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
  invalid={false}
  isIncomplete={false}
  options={Array []}
  required={false}
/>
`;

exports[`NativeSelect should render a NativeSelect with onChange 1`] = `
<NativeSelect
  disabled={false}
  id="test-id"
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
  invalid={false}
  isIncomplete={false}
  onChange={[MockFunction]}
  options={Array []}
  required={false}
/>
`;

exports[`NativeSelect should render a NativeSelect with opt groups 1`] = `
<NativeSelect
  disabled={false}
  id="test-id"
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
  invalid={false}
  isIncomplete={false}
  options={
    Array [
      Object {
        "childOptions": Array [
          Object {
            "display": "Volvo",
            "value": "volvo",
          },
          Object {
            "display": "Saab",
            "value": "saab",
          },
        ],
        "display": "Swedish Cars",
      },
      Object {
        "childOptions": Array [
          Object {
            "display": "Mercedes",
            "value": "mercedes",
          },
          Object {
            "display": "Audi",
            "value": "audi",
          },
        ],
        "display": "German Cars",
      },
    ]
  }
  required={false}
/>
`;

exports[`NativeSelect should render a NativeSelect with options and defaultValue 1`] = `
<NativeSelect
  defaultValue="saab"
  disabled={false}
  id="test-id"
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
  invalid={false}
  isIncomplete={false}
  options={
    Array [
      Object {
        "display": "Volvo",
        "value": "volvo",
      },
      Object {
        "display": "Saab",
        "value": "saab",
      },
      Object {
        "display": "Mercedes",
        "value": "mercedes",
      },
      Object {
        "display": "Audi",
        "value": "audi",
      },
    ]
  }
  required={false}
/>
`;

exports[`NativeSelect should render a NativeSelect with options and value 1`] = `
<NativeSelect
  disabled={false}
  id="test-id"
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
  invalid={false}
  isIncomplete={false}
  options={
    Array [
      Object {
        "display": "Volvo",
        "value": "volvo",
      },
      Object {
        "display": "Saab",
        "value": "saab",
      },
      Object {
        "display": "Mercedes",
        "value": "mercedes",
      },
      Object {
        "display": "Audi",
        "value": "audi",
      },
    ]
  }
  required={false}
  value="audi"
/>
`;

exports[`NativeSelect should render a default NativeSelect 1`] = `
<NativeSelect
  disabled={false}
  id="test-id"
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
  invalid={false}
  isIncomplete={false}
  options={Array []}
  required={false}
/>
`;
